extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -386,16 +385,58 @@ public void run()
             records = consumer.poll(POLL_TIMEOUT);
           }
           catch (OffsetOutOfRangeException e) {
-            log.warn("OffsetOutOfRangeException with message [%s], retrying in %dms", e.getMessage(), POLL_RETRY_MS);
-            pollRetryLock.lockInterruptibly();
-            try {
-              long nanos = TimeUnit.MILLISECONDS.toNanos(POLL_RETRY_MS);
-              while (nanos > 0L && !pauseRequested && !stopRequested) {
-                nanos = isAwaitingRetry.awaitNanos(nanos);
+            // Reset consumer offset if resetOffsetAutomatically is set to true
+            // and the current message offset in the kafka partition is more than the
+            // next message offset that we are trying to fetch
+            if (tuningConfig.isResetOffsetAutomatically()) {
```

Comment:
Can all this logic be moved into a method to keep the main while loop as clean as possible?